### PR TITLE
feat(journey-maps): points of interest environment option

### PR DESCRIPTION
**Pull request by besartmemeti**

### CODE REVIEW

**Comment by besartmemeti** [comment]
Info for Rokas-members: to test the environment, check out the following coordinate:

[
      8.54225,
      47.37953
]

**Comment by sebisteiner** [comment]
> Preview ready from [a0621e8](https://github.com/sbb-design-systems/sbb-angular/commit/a0621e8cc45afe147ebbaeca113a987adf48e919) at 30.9.2022, 08:34:42:
> 
> * https://preview-sbb-angular.app.sbb.ch/1679

The preview is broken... I think @swiss-chris deleted the API key 🥲 I'll send a new one to @kyubisation.

**Comment by swiss-chris** [comment]
> > Preview ready from [a0621e8](https://github.com/sbb-design-systems/sbb-angular/commit/a0621e8cc45afe147ebbaeca113a987adf48e919) at 30.9.2022, 08:34:42:
> > 
> > * https://preview-sbb-angular.app.sbb.ch/1679
> 
> The preview is broken... I think @swiss-chris deleted the API key 🥲 I'll send a new one to @kyubisation.

Falls ich es deleted hab, müsste es hier drin sein: https://confluence.sbb.ch/pages/viewpage.action?pageId=2050923409

**Comment by sebisteiner** [comment]
> > > Preview ready from [a0621e8](https://github.com/sbb-design-systems/sbb-angular/commit/a0621e8cc45afe147ebbaeca113a987adf48e919) at 30.9.2022, 08:34:42:
> > > 
> > > * https://preview-sbb-angular.app.sbb.ch/1679
> > 
> > 
> > The preview is broken... I think @swiss-chris deleted the API key 🥲 I'll send a new one to @kyubisation.
> 
> Falls ich es deleted hab, müsste es hier drin sein: https://confluence.sbb.ch/pages/viewpage.action?pageId=2050923409

Sorry - my bad... I think my problems were due to my ModHeader Chrome plugin.

File: src/journey-maps/angular/journey-maps.ts [hunk 1]
```diff
@@ -501,9 +501,23 @@ export class SbbJourneyMaps implements OnInit, AfterViewInit, OnDestroy, OnChang
...
-        this._mapService.updatePoiVisibility(this._map, this.poiOptions)
-      );
+      this._executeWhenMapStyleLoaded(() => {
+        const poiEnvironmentChanged =
+          !changes.poiOptions.firstChange &&
+          changes.poiOptions.previousValue?.environment !==
+            changes.poiOptions.currentValue?.environment;
+
+        if (poiEnvironmentChanged) {
+          // Wait for Visibility to switch until new style got loaded
```
Comment (sebisteiner) [inline src/journey-maps/angular/journey-maps.ts:511]: In my opinion it's not necessary to reload the whole style. We also could replace the URL of the existing source.

    const source = this._map.getSource('journey-pois-source') as VectorTileSource;
    // PROD --> INT
    source.setUrl(source.url.replace('journey_pois', 'journey_pois_integration'));

**Review by sebisteiner — CHANGES_REQUESTED** [review]

File: src/journey-maps/angular/journey-maps.ts (hunk 1 above)
Comment (besartmemeti) [reply src/journey-maps/angular/journey-maps.ts]: Implemented in newest commit.

File: src/journey-maps/angular/journey-maps.interfaces.ts [hunk 2]
```diff
@@ -183,6 +183,8 @@ export type SbbTemplateType = TemplateRef<any> | string;
 export interface SbbPointsOfInterestOptions {
   /** Configure a list of points of interest categories visible in the map. Set empty, to hide all POIs. */
   categories: SbbPointsOfInterestCategoryType[];
+  /** Configure the environment from which to get the POIs from (default = PROD). */
+  environment?: SbbPointsOfInterestEnvironmentType;
```
[the remark this answers is not in the document]
Comment (besartmemeti) [reply src/journey-maps/angular/journey-maps.interfaces.ts:187]: Implemented in newest commit.

**Review by sebisteiner — APPROVED** [review]
LGTM 👍

**Pull request merged**